### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by gauravpatil123**

### CODE REVIEW

File: DataPreprocessing.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
```
Comment (croc-ctrl) [inline DataPreprocessing.py:1]: The code here is too free form, will be good to wrap this into better objects or functions

File: DataPreprocessing.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        n_gram_sequence = token_list[:i+1]
+        input_sequences.append(n_gram_sequence)
+
+#print(input_sequences[:150])
+
+max_sequence_length = max([len(x) for x in input_sequences])
+input_sequences = np.array(pad_sequences(input_sequences, maxlen = max_sequence_length, padding = 'pre'))
+
+predictors, label = input_sequences[:,:-1], input_sequences[:,-1]
+label = ku.to_categorical(label, num_classes = total_words)
```
Comment (croc-ctrl) [inline DataPreprocessing.py]: What happens to labels after this step?

File: DataPreprocessing.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+"""
+
+from tensorflow.keras.preprocessing.sequence import pad_sequences
+from tensorflow.keras.preprocessing.text import Tokenizer
+import tensorflow.keras.utils as ku
+import numpy as np
+
+tokenizer = Tokenizer()
+
+data = open('data/data.txt').read()
```
Comment (croc-ctrl) [inline DataPreprocessing.py]: This should ideally be provided as a function argument

**Review by croc-ctrl — CHANGES_REQUESTED** [review]
Its a good first step

File: DataPreprocessing.py (hunk 1 above)
Comment (gauravpatil123) [reply DataPreprocessing.py:1]: Addressed this issue in the new branch (see new pull req)

File: DataPreprocessing.py (hunk 2 above)
Comment (gauravpatil123) [reply DataPreprocessing.py]: changed the data structure so no longer applies, (for the previous code they were supposed to be used dynamically in Train and predict scripts)
